### PR TITLE
dry module loading

### PR DESCRIPTION
**Pull request by mmatera**
The idea is to avoid to list the builtin modules two times...

### CODE REVIEW

**Comment by rocky** [comment]
> The idea is to avoid to list the builtin modules two times...

I do - the idea is to introspect on the module. The Python inspect module/library describes this and has functions for it. 

And I do something like this in my debuggers.

File: mathics/builtin/__init__.py [hunk 1]
```diff
@@ -1,33 +1,88 @@
...
-    plot, physchemdata, randomnumbers, recurrence, specialfunctions, scoping,
-    strings, structure, system, tensors, xmlformat, optimization)
+if ENABLE_FILES_MODULE:
+    # from mathics.builtin import files, importexport
+    module_names += ["files", "importexport"]
+
+modules = []
+
+for module_name in module_names:
+    exec(f"from mathics.builtin import {module_name}")
```
Comment (rocky) [inline mathics/builtin/__init__.py:58]: I think importlib is what you want to use here instead of exec

Comment (mmatera) [reply mathics/builtin/__init__.py:58]: I'm not sure how to implement the equivalent of the line 58 with importlib. How would you do that?

Comment (rocky) [reply mathics/builtin/__init__.py:58]: https://github.com/rocky/python3-trepan/blob/master/trepan/processor/cmdproc.py#L1062

**Comment by rocky** [comment]
@mmatera if you have your copy around - please `git push --force origin HEAD` on top of this.

I should have branched on top of this rather than commiting on top. 

The importlib code was largely correct. It worked on my setup. However there may be some interaction with doc testing I haven't worked out.

**Comment by mmatera** [comment]
> @mmatera if you have your copy around - please `git push --force origin HEAD` on top of this.
> 
> I should have branched on top of this rather than commiting on top.
> 
> The importlib code was largely correct. It worked on my setup. However there may be some interaction with doc testing I haven't worked out.

OK, no problem: I keep my local version.

> @mmatera if you have your copy around - please `git push --force origin HEAD` on top of this.
> 
> I should have branched on top of this rather than commiting on top.
> 
> The importlib code was largely correct. It worked on my setup. However there may be some interaction with doc testing I haven't worked out.

Done.

**Comment by rocky** [comment]
Superceded by #1138

**Pull request closed without merging**